src/main/java/picard/analysis/StrandSpecificityUtil.java

```diff
@@ -0,0 +1,196 @@
+            else if (rec.getReadUnmappedFlag() || (rec.getReadPairedFlag() && rec.getMateUnmappedFlag())) continue;
+            else if (rec.getMappingQuality() < minimumMappingQuality) continue;
+            else if (rec.getDuplicateReadFlag()) continue;
+            else if (rec.getReadPairedFlag() && SamPairUtil.getPairOrientation(rec) != SamPairUtil.PairOrientation.FR) continue;
+
+            // Grab information about the alignment and overlapping genes etc.
+            final Interval readInterval = new Interval(rec.getReferenceName(), rec.getAlignmentStart(), rec.getAlignmentEnd());
+            final Collection<Gene> overlappingGenes = geneOverlapDetector.getOverlaps(readInterval);
+
+            // Ignore reads that overlap multiple genes
+            if (overlappingGenes.size() == 0) continue;
+
+            // Find if any alignment block overlaps a UTR or coding region in any gene
+            final List<AlignmentBlock> alignmentBlocks = rec.getAlignmentBlocks();
+            boolean overlapsExon = alignmentBlocks.stream().anyMatch(alignmentBlock -> {
+                // Get functional class for each position in the alignment block.
+                final LocusFunction[] locusFunctions = new LocusFunction[alignmentBlock.getLength()];
+
+                // By default, if base does not overlap with rRNA or gene, it is intergenic.
+                Arrays.fill(locusFunctions, 0, locusFunctions.length, LocusFunction.INTERGENIC);
+
+                // Assign the function for each transcripts
+                for (final Gene gene : overlappingGenes) {
+                    for (final Gene.Transcript transcript : gene) {
+                        transcript.assignLocusFunctionForRange(alignmentBlock.getReferenceStart(), locusFunctions);
+                    }
+                }
+
+                // Tally the function of each base in the alignment block.
+                for (final LocusFunction locusFunction : locusFunctions) {
+                    switch (locusFunction) {
+                        case UTR:
+                        case CODING:
+                            return true;
+                        case INTERGENIC:
+                        case INTRONIC:
+                        case RIBOSOMAL:
+                            break;
+                    }
+                }
+
+                return false;
+            });
+
+            // Only examine reads that overlap an exon
+            if (overlapsExon) {
+                // We would need both read pairs to check multiple genes, and therefore we ignore such cases and call
+                // them unclassified later.
+                if (overlappingGenes.stream().map(Gene::isPositiveStrand).distinct().count() == 1) {
+                    final boolean negativeReadStrand            = rec.getReadNegativeStrandFlag();
+                    final boolean negativeTranscriptionStrand   = overlappingGenes.iterator().next().isNegativeStrand();
```

Comment:
I see above that you are limiting to where overlapped genes are on the same strand.  Maybe comment there why, and that is makes this safe?